yt_dlp/extractor/tviplayer.py

```diff
@@ -37,12 +38,26 @@ class TVIPlayerIE(InfoExtractor):
             'duration': 148,
             'season_number': 2,
         }
+    }, {
+        # episodio url
+        'url': 'https://tviplayer.iol.pt/programa/para-sempre/61716c360cf2365a5ed894c4/episodio/t1e187',
+        'info_dict': {
+            'id': 't1e187',
+            'ext': 'mp4',
+            'season': 'Season 1',
+            'title': 'Quem denunciou Pedro?',
+            'thumbnail': 'https://www.iol.pt/multimedia/oratvi/multimedia/imagem/id/62eda30b0cf2ea367d48973b/',
+            'duration': 1250,
+            'season_number': 1,
+        }
     }]
+    _TOKEN = None
 
     def _real_initialize(self):
-        self.wms_auth_sign_token = self._download_webpage(
-            'https://services.iol.pt/matrix?userId=', 'wmsAuthSign',
-            note='Trying to get wmsAuthSign token')
+        if TVIPlayerIE._TOKEN is None:
+            TVIPlayerIE._TOKEN = self._download_webpage(
+                'https://services.iol.pt/matrix?userId=', 'wmsAuthSign',
+                note='Trying to get wmsAuthSign token')
```

Comment:
real_initialize only ever gets called once. Could you explain the reasoning for this change?

Reply:
It just to not recalling the token in test. Before this commit, it always try call to this token for every test case. 

<details> <summary> log before this commit</summary>

```shell
(python_proj) ~/.../python_proj/yt-dlp $ python test/test_download.py TestDownload.test_TVIPlayer_all
[TVIPlayer] wmsAuthSign: Trying to get wmsAuthSign token
[debug] [TVIPlayer] Extracting URL: https://tviplayer.iol.pt/programa/jornal-das-8/53c6b3903004dc006243d0cf/video/61c8e8b90cf2c7ea0f0f71a9
[TVIPlayer] 61c8e8b90cf2c7ea0f0f71a9: Downloading webpage
[TVIPlayer] 61c8e8b90cf2c7ea0f0f71a9: Downloading m3u8 information
[info] 61c8e8b90cf2c7ea0f0f71a9: Downloading 1 format(s): 2000
[info] Writing video metadata as JSON to: test_TVIPlayer_61c8e8b90cf2c7ea0f0f71a9.info.json
[debug] Invoking hlsnative downloader on "https://streaming-vod1.iol.pt/vod/_definst_/7/1/a/9/smil:61c8e8b90cf2c7ea0f0f71a9-L/chunklist_w1349536125_b2000000.m3u8?nimblesessionid=549146125&wmsAuthSign=c2VydmVyX3RpbWU9OC83LzIwMjIgMjoyNjo1OSBQTSZoYXNoX3ZhbHVlPUgvL2dkaTBianVTMEZJL1g1eEs4OEE9PSZ2YWxpZG1pbnV0ZXM9MTQ0MCZpZD1mNzc1ZjgzMi0xNDBmLTQ1MjQtYWQ4MS01NDcyNWMyNjQ1NDQ="
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 417
[download] Destination: test_TVIPlayer_61c8e8b90cf2c7ea0f0f71a9.mp4
[download]   0.0% of ~1.07GiB at    690.55B/s ET[download]   0.0% of ~1.07GiB at    2.02KiB/s ET[download]   0.0% of ~1.07GiB at    4.70KiB/s ET[download]   0.0% of ~1.07GiB at   10.04KiB/s ET[download]   0.0% of ~1.07GiB at   20.71KiB/s ET[download]   0.0% of ~1.07GiB at   36.13KiB/s ET[download]   0.0% of ~1.07GiB at   63.40KiB/s ET[download]   0.0% of ~1.07GiB at  111.81KiB/s ET[download]   0.0% of ~1.07GiB at  199.86KiB/s ET[download]   0.1% of ~1.07GiB at  355.82KiB/s ET[download]   0.2% of ~1.07GiB at  592.56KiB/s ET[download]   0.2% of ~1.07GiB at  711.98KiB/s ET[download]   0.2% of ~1.07GiB at  711.33KiB/s ET[download] 100% of 2.62MiB in 00:03 at 707.26KiB/s
[TVIPlayer] wmsAuthSign: Trying to get wmsAuthSign token
[debug] [TVIPlayer] Extracting URL: https://tviplayer.iol.pt/programa/isabel/62b471090cf26256cd2a8594/video/62be445f0cf2ea4f0a5218e5
[TVIPlayer] 62be445f0cf2ea4f0a5218e5: Downloading webpage
[TVIPlayer] 62be445f0cf2ea4f0a5218e5: Downloading m3u8 information
[info] 62be445f0cf2ea4f0a5218e5: Downloading 1 format(s): 2000
[info] Writing video metadata as JSON to: test_TVIPlayer_1_62be445f0cf2ea4f0a5218e5.info.json
[debug] Invoking hlsnative downloader on "https://streaming-vod1.iol.pt/vod/_definst_/1/8/e/5/smil:62be445f0cf2ea4f0a5218e5-L/chunklist_w522640482_b2000000.m3u8?nimblesessionid=549146225&wmsAuthSign=c2VydmVyX3RpbWU9OC83LzIwMjIgMjoyNzoxMSBQTSZoYXNoX3ZhbHVlPVdpb0hkRHY0MTVZZndyM3drRUdjaEE9PSZ2YWxpZG1pbnV0ZXM9MTQ0MCZpZD1jZDJhOWExMi0wNGVjLTQ4YTYtYjA0Ni1jMGNjYzliZDI5ODM="
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 326
[download] Destination: test_TVIPlayer_1_62be445f0cf2ea4f0a5218e5.mp4
[download]   0.0% of ~686.71MiB at    749.13B/s [download]   0.0% of ~686.71MiB at    2.18KiB/s [download]   0.0% of ~686.71MiB at    5.06KiB/s [download]   0.0% of ~686.71MiB at   10.78KiB/s [download]   0.0% of ~686.71MiB at   22.15KiB/s [download]   0.0% of ~686.71MiB at   38.89KiB/s [download]   0.0% of ~686.71MiB at   67.62KiB/s [download]   0.0% of ~686.71MiB at  119.59KiB/s [download]   0.1% of ~686.71MiB at  211.78KiB/s [download]   0.1% of ~686.71MiB at  376.96KiB/s [download]   0.3% of ~686.71MiB at  616.09KiB/s [download]   0.3% of ~686.71MiB at  646.67KiB/s [download]   0.3% of ~686.71MiB at  645.90KiB/s [download] 100% of 2.11MiB in 00:03 at 642.33KiB/s
[TVIPlayer] wmsAuthSign: Trying to get wmsAuthSign token
[debug] [TVIPlayer] Extracting URL: https://tviplayer.iol.pt/video/62c4131c0cf2f9a86eac06bb
[TVIPlayer] 62c4131c0cf2f9a86eac06bb: Downloading webpage
[TVIPlayer] 62c4131c0cf2f9a86eac06bb: Downloading m3u8 information
[info] 62c4131c0cf2f9a86eac06bb: Downloading 1 format(s): 2000
[info] Writing video metadata as JSON to: test_TVIPlayer_2_62c4131c0cf2f9a86eac06bb.info.json
[debug] Invoking hlsnative downloader on "https://streaming-vod2.iol.pt/vod/_definst_/0/6/b/b/smil:62c4131c0cf2f9a86eac06bb-L/chunklist_w919645678_b2000000.m3u8?nimblesessionid=233408121&wmsAuthSign=c2VydmVyX3RpbWU9OC83LzIwMjIgMjoyNzoyMyBQTSZoYXNoX3ZhbHVlPU9jQ01LWGE1akgvenNvbjRFcmlSYkE9PSZ2YWxpZG1pbnV0ZXM9MTQ0MCZpZD03ZGQwOGNkMC1iMDI5LTQyMzUtYTk3ZC05ODZkMDBjYzQyNTI="
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 15
[download] Destination: test_TVIPlayer_2_62c4131c0cf2f9a86eac06bb.mp4
[download]   0.0% of ~43.82MiB at    760.92B/s E[download]   0.0% of ~43.82MiB at    2.22KiB/s E[download]   0.0% of ~43.82MiB at    5.16KiB/s E[download]   0.0% of ~43.82MiB at   11.01KiB/s E[download]   0.1% of ~43.82MiB at   22.64KiB/s E[download]   0.1% of ~43.82MiB at   39.19KiB/s E[download]   0.3% of ~43.82MiB at   68.25KiB/s E[download]   0.6% of ~43.82MiB at  120.07KiB/s E[download]   1.1% of ~43.82MiB at  189.53KiB/s E[download]   2.1% of ~43.82MiB at  324.20KiB/s E[download]   4.1% of ~43.82MiB at  509.87KiB/s E[download]   6.7% of ~43.82MiB at  673.00KiB/s E[download]   6.7% of ~43.82MiB at  672.59KiB/s E[download] 100% of 2.92MiB in 00:04 at 669.31KiB/s
.
----------------------------------------------------------------------
Ran 1 test in 40.163s

OK
```
</details>

I don't mind revert this if you mind about this commit

Reply:
It sounds the the issue is in the test runner. Let me have a look

Reply:
In the mean time, I have cherrypicked just the other patch